src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,128 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] elementsData;
+    private Object[] emptyArray = {};
+    private int size;
+
+    public ArrayList() {
+        elementsData = emptyArray;
+        size = 0;
+    }
+
+    public ArrayList(int arraySize) {
+        size = 0;
+        if (arraySize > 0) {
+            elementsData = new Object[arraySize];
+        } else if (arraySize == 0) {
+            elementsData = emptyArray;
+        } else {
+            throw new IllegalArgumentException("Incorrect size of array");
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementsData.length) {
+            elementsData = grow();
+        }
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        if (size == elementsData.length) {
+            elementsData = grow();
+        }
+        System.arraycopy(elementsData, index, elementsData, index + 1, size - index);
+        elementsData[index] = value;
+        size += 1;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            return;
+        }
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
foreach might look better :)

Reply:
I can't understand how to do it. May I get a hint?
![fore](https://user-images.githubusercontent.com/17779779/70704082-fb6e1d00-1cd9-11ea-8477-731508d87b6e.jpg)

Reply:
well, I had a try, and it seems to be not really nice. sorry, my bad :( your solution is good